app/models/labware_creators/plate_split_to_tube_racks.rb

```diff
@@ -0,0 +1,453 @@
+  # optional).
+  #
+  # The parent plate contains multiple copies of material prepared from the same sample. One well instance
+  # of each sample will go into a tube in the 'sequencing' rack (if present), and any remaining copies
+  # will go into tubes in the 'contingency' rack.
+  #
+  # If after the initial preparation the users feel they need more contingency tubes, then they will go
+  # back to prepare more material for that sample from an earlier step in the pipeline and create just
+  # 'contingency' tubes at this step.
+  #
+  # Inputs:
+  # 1) The parent plate - contains a number of wells containing material prepared from the same sample.
+  #    The first of these parent wells will be transferred into a tube in the 'sequencing' rack if it is
+  #    present, and any remaining parent wells for the same sample will be transferred into tubes in the
+  #    'contingency' rack.
+  # 2) Child tube rack scan CSV files - these are scans of racks of 2D tube barcodes from the 'contingency'
+  #    and 'sequencing' tube racks, to allow us to know the position and barcode of each available tube.
+  #    On the upload screen displays counts of how many tubes are needed (e.g. To perform this transfer you
+  #    will either need 20 sequencing and 40 contingency tubes, or 60 contingency tubes.)
+  #
+  # Validations - Error message to users if any of these are not met:
+  # 1) The user must always upload a scan file for the 'contingency' rack tube barcodes, whereas the
+  #    'sequencing' rack file is optional.
+  # 2) The scanned child tube barcodes must be unique and must not already exist in the system i.e. they are
+  #    new unused empty tubes. List any that do already exist with rack type and location.
+  # 3) The number of tubes available in the racks must be sufficient for the number of parent wells being
+  #    transferred. e.g. if there are 20 distinct samples in the parent and 40 additional copies (60 wells
+  #    total), then there must be at least 20 tubes in the first rack and at least 40 in the second rack.
+  #    List any racks that do not have enough tubes with totals.
+  #
+  # rubocop:disable Metrics/ClassLength
+  class PlateSplitToTubeRacks < Base
+    include SupportParent::PlateOnly
+
+    self.attributes += %i[sequencing_file contingency_file]
+
+    self.page = 'plate_split_to_tube_racks'
+
+    attr_accessor :sequencing_file, :contingency_file
+
+    attr_reader :child_sequencing_tubes, :child_contingency_tubes
+
+    validates_nested :well_filter
+
+    # TODO: can add validations for specific metadata fields here if needed
+
+    # Don't create the tubes until at least the contingency file has been uploaded
+    validates :contingency_file, presence: true
+
+    # N.B. contingency file is required, sequencing file is optional
+    validates_nested :sequencing_csv_file, if: :contingency_file
+    validates_nested :contingency_csv_file, if: :sequencing_file
```

Comment:
Are these meant to be swapped round? validates sequencing file if contingency is uploaded, and vice vera? Probably just misunderstanding something!